toolchain/lexer/token_kind.h

```diff
@@ -83,4 +78,18 @@ class TokenKind : public CARBON_ENUM_BASE(TokenKind) {
 
 }  // namespace Carbon
 
+namespace llvm {
+
+// We use formatv primarily for diagnostics. In these cases, it's expected that
+// the spelling in source code should be used.
+template <>
+struct format_provider<Carbon::TokenKind> {
```

Comment:
The goal here makes perfect sense.

Using the format provider in this way feels pretty brittle though.... I feel like we should have a diagnostic specific extension point instead. But seems fine to just leave a TODO here for now? Let's the cleanups here land and we can switch to a dedicated extension point in a follow-up?